pkg/api/persistentvolumeclaim/util.go

```diff
@@ -151,3 +154,18 @@ func allocatedResourcesInUse(oldPVC *core.PersistentVolumeClaim) bool {
 
 	return false
 }
+
+func GetWarningsForPersistentVolumeClaim(pv *core.PersistentVolumeClaim) []string {
+	if pv == nil {
+		return nil
+	}
+	storageValue := pv.Spec.Resources.Requests[core.ResourceStorage]
+	var warnings []string
+	if storageValue.MilliValue()%int64(1000) != int64(0) {
+		warnings = append(warnings, fmt.Sprintf(
+			"%s: fractional byte value %q is invalid, must be an integer",
+			field.NewPath("spec").Child("resources").Child("requests").Key(core.ResourceStorage.String()),
+			storageValue.String()))
+	}
```

Comment:
wondering if this bit should be factored into a GetWarningsForPersistentVolumeClaimSpec function we can call when getting warnings for statefulsets and podspecs

Reply:
Good point. I thought this is not used by other functions and merged the two functions together. If there are other usages, it should be refactored.